File: Deepseek-Coder-33b-Instruct/stage2/acc_update_self.cpp
#include <iostream>
#include <openacc.h>

#ifndef T1
//T1:acc update self,V:2.7-3.3
int test1(){
    int err = 0;
    const int size = 100;
    int data[size];

    // Initialize data on the host
    for(int i = 0; i < size; i++) {
        data[i] = i;
    }

    // Copy data to the device
    #pragma acc enter data copyin(data[0:size])

    // Update data on the device
    #pragma acc update self(data[0:size])

    // Check if the update was successful
    for(int i = 0; i < size; i++) {
        if(data[i] != i) {
            err = 1;
            break;
        }
    }

    // Copy data back to the host
    #pragma acc exit data delete(data[0:size])

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < 1; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/acc_update_self.f90
!T1:acc update self,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        REAL(KIND=4), DIMENSION(N) :: a, b
        INTEGER :: i

        ! Initialize array
        DO i = 1, N
          a(i) = REAL(i)
        END DO

        ! Copy array to device
        !$acc data copyin(a) copyout(b)

        ! Update array on device
        !$acc update device(a)

        ! Update array on host
        !$acc update self(a)

        !$acc end data

        ! Check if array was updated correctly
        DO i = 1, N
          IF (a(i) .ne. REAL(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM